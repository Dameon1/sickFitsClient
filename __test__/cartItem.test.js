import wait from 'waait';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import CartItem from '../components/CartItem';
import { fakeCartItem } from '../lib/testUtils';
import { mount } from 'enzyme';

const fakeItem = fakeCartItem();

describe('<CartItem/>', () => {
  it('renders and matches snapshot', async () => {
    const wrapper = mount(
        <MockedProvider >
          <CartItem cartItem={fakeItem}/>
        </MockedProvider>
      );
       await wait();
       wrapper.update();
       expect(toJSON(wrapper.find('CartItem'))).toMatchSnapshot();
       expect(wrapper.find('CartItem')).toHaveLength(1);
      });
});