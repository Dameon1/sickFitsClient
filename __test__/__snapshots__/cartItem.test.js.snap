// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem/> renders and matches snapshot 1`] = `
<CartItem
  cartItem={
    Object {
      "__typename": "CartItem",
      "id": "omg123",
      "item": Object {
        "__typename": "Item",
        "description": "dogs",
        "id": "abc123",
        "image": "dog-small.jpg",
        "largeImage": "dog.jpg",
        "price": 5000,
        "title": "dogs are best",
        "user": null,
      },
      "quantity": 3,
      "user": Object {
        "__typename": "User",
        "cart": Array [],
        "email": "Delmer.Smith@yahoo.com",
        "id": "4234",
        "name": "Miss Coleman Berge",
        "orders": Array [],
        "permissions": Array [
          "ADMIN",
        ],
      },
    }
  }
>
  <CartItem__CartItemStyles>
    <li
      className="CartItem__CartItemStyles-sc-1rm9l7o-0 hQBQSZ"
    >
      <img
        alt="dogs are best"
        src="dog-small.jpg"
      />
      <div
        className="cart-item-detaitls"
      >
        <h3>
          dogs are best
        </h3>
        <p>
          $150
           - 
          <em>
            3
             × 
            $50
          </em>
        </p>
      </div>
      <RemoveFromCart
        id="omg123"
      >
        <Mutation
          mutation={
            Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "removeFromCart",
                  },
                  "operation": "mutation",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "value": Object {
                              "kind": "Variable",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                            },
                          },
                        ],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "removeFromCart",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [
                    Object {
                      "defaultValue": undefined,
                      "kind": "VariableDefinition",
                      "type": Object {
                        "kind": "NonNullType",
                        "type": Object {
                          "kind": "NamedType",
                          "name": Object {
                            "kind": "Name",
                            "value": "ID",
                          },
                        },
                      },
                      "variable": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                      },
                    },
                  ],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 88,
                "start": 0,
              },
            }
          }
          optimisticResponse={
            Object {
              "__typename": "Mutation",
              "removeFromCart": Object {
                "__typename": "CartItem",
                "id": "omg123",
              },
            }
          }
          update={[Function]}
          variables={
            Object {
              "id": "omg123",
            }
          }
        >
          <RemoveFromCart__BigButton
            disabled={false}
            onClick={[Function]}
            title="Delete Item"
          >
            <button
              className="RemoveFromCart__BigButton-emvtd6-0 gPWNSn"
              disabled={false}
              onClick={[Function]}
              title="Delete Item"
            >
              ×
            </button>
          </RemoveFromCart__BigButton>
        </Mutation>
      </RemoveFromCart>
    </li>
  </CartItem__CartItemStyles>
</CartItem>
`;
